ui/ozone/platform/wayland/wayland_keyboard.cc

```diff
@@ -108,8 +109,15 @@ void WaylandKeyboard::Key(void* data,
           dom_code, flags, &dom_key, &key_code))
     return;
 
-  // TODO(tonikitoo): handle repeat here.
   bool down = state == WL_KEYBOARD_KEY_STATE_PRESSED;
+
+  // TODO(tonikitoo): handle repeat here.
+  bool repeat = false;
+  if (!repeat) {
+    int flag = ModifierDomKeyToEventFlag(dom_key);
+    keyboard->UpdateModifier(flag, down);
```

Comment:
Why do you update modifiers here? WaylandKeyboard::Modifiers should send all the possible modifiers, IIRC.

Reply:
This is because when a modifier key is pressed, WaylandKeyboard::Key is called before WaylandKeyboard::Modifiers. KeyboardEvDev also does this to prevent such case:

https://cs.chromium.org/chromium/src/ui/events/ozone/evdev/keyboard_evdev.cc?q=keyboardevdev&sq=package:chromium&dr=CSs&l=238.

This is the precise line the fixes Issue #331.

Reply:
Got it! Thanks